### PR TITLE
Added a way to override switch configuration based on the request parameters

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Override switch configuration based on the request parameter.
Like you want to enable externalportal for a specific ssid and not for another one for the same AP or select a different switch module based on uri parameters.

# Impacts
- RADIUS authorize workflow
- External Protal workflow

# Delete branch after merge
YES

# NEWS file entries
## New Features
* Switch filter to allow to averride switch configuration based on the request

### CODE REVIEW

File: lib/pf/access_filter/switch.pm [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+use Scalar::Util qw(reftype);
+
+use base qw(pf::access_filter);
+tie our %ConfigSwitchFilters, 'pfconfig::cached_hash', 'config::SwitchFilters';
+tie our %SwitchFilterEngineScopes, 'pfconfig::cached_hash', 'FilterEngine::SwitchScopes';
+
+=head2 filterRule
+
+    Handle the role update
```
Comment (louismunro) [inline lib/pf/access_filter/switch.pm]: The role update?

File: lib/pf/pfcmd/checkup.pm [hunk 2]
```diff
@@ -1050,7 +1050,7 @@ sub vlan_filter_rules {
             add_problem ( $WARN, "Missing operator attribute in $rule vlan filter rule")
                 if (!defined($ConfigVlanFilters{$rule}->{'operator'}));
             add_problem ( $WARN, "Missing value attribute in $rule vlan filter rule")
-                if (!defined($ConfigVlanFilters{$rule}->{'value'}));
+                if (!defined($ConfigVlanFilters{$rule}->{'value'}) && $ConfigVlanFilters{$rule}->{'operator'} ne 'defined' && $ConfigVlanFilters{$rule}->{'operator'} ne 'not_defined');
```
Comment (louismunro) [inline lib/pf/pfcmd/checkup.pm:1053]: Should this be in this pull request?
It does not seem to be relevant to switch_filters, unless I am mistaken.

**Comment by louismunro** [comment]
LGTM.

**Comment by dwlfrth** [comment]
@fdurand please adjust and let's merge
